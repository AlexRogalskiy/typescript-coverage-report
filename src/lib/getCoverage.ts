import {
  lint,
  AnyInfo,
  FileTypeCheckResult,
  LintOptions
} from "type-coverage-core";

type FileCount = Map<
  string,
  Pick<FileTypeCheckResult, "correctCount" | "totalCount">
>;

export type CoverageData = {
  fileCounts: FileCount;
  anys: AnyInfo[];
  percentage: number;
  total: number;
  covered: number;
  uncovered: number;
};

const getCoverage = async (
  options?: Pick<LintOptions, "strict" | "debug">
): Promise<CoverageData> => {
  const { anys, fileCounts, totalCount, correctCount } = await lint(".", {
    ...options,
    fileCounts: true
  });

  return {
    fileCounts,
    anys,
    percentage: (correctCount * 100) / totalCount,
    total: totalCount,
    covered: correctCount,
    uncovered: totalCount - correctCount
  };
};

export default getCoverage;
