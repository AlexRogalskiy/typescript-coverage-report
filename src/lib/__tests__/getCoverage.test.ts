import getCoverage from "../getCoverage";

describe("getCoverage function", () => {
  it("returns calculated data from type-coverage-core", async (done) => {
    const data = await getCoverage();

    expect(data).toMatchSnapshot();

    done();
  });
});
