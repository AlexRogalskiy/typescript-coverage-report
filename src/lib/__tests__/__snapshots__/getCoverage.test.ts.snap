// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCoverage function returns calculated data from type-coverage-core 1`] = `
Object {
  "anys": Array [],
  "covered": 100,
  "fileCounts": Map {
    "index.html" => Object {
      "correctCount": 100,
      "totalCount": 100,
    },
  },
  "percentage": 100,
  "total": 100,
  "uncovered": 0,
}
`;
